x-pack/legacy/plugins/integrations_manager/common/types.ts

```diff
@@ -10,14 +10,14 @@ export { Request, ResponseToolkit, Server, ServerRoute } from 'hapi';
 
 export type InstallationStatus = Installed['status'] | NotInstalled['status'];
 
-export type AssetType =
-  | 'config'
-  | 'dashboard'
-  | 'index-pattern'
-  | 'ingest-pipeline'
-  | 'search'
-  | 'timelion-sheet'
-  | 'visualization';
+export enum AssetType {
```

Comment:
I hope this works. It's what I wanted to do initially, but had some challenges. I think I've both fixed the things that were getting in the way and have a better hand on TS but I'll run locally to confirm. 🤞

Reply:
Ran locally and things still work as expected. I was struggling with a way to have these strings as both a union type & some way to do "lookup" for the "does type x use saved objects?" question. I ended up using a `Set` for `SAVED_OBJECT_TYPES` and we can create that from a few different structures (this PR shows that) so seems ok.

Can you talk about why you changed it to an `enum`? Is it solely to be able to use code like `AssetType.config` (and have one linked property name vs N duplicated strings) or are there other benefits?

FWIW, I think I'm ok changing to this. I'm just documenting what I can remember about why I went one way, and I'd like to know more about the benefits and trade-offs. I'm still new to TS & enum so I don't have practical experience with any benefits/drawbacks.

Reply:
I wanted to be able to iterate over them, even if in the end I didn't need to. Also, for me it feels like the correct type -- we use them in the metrics UI for things like node types so I got used to using them there.